contracts/dapis/DapiServer.sol

```diff
@@ -495,6 +495,40 @@ contract DapiServer is
         emit UpdatedBeaconWithSignedData(beaconId, decodedData, timestamp);
     }
 
+    /// @notice Updates a Beacon using data signed for this contract by the
+    /// respective Airnode, without requiring a request or subscription
+    /// @param airnode Airnode address
+    /// @param templateId Template ID
+    /// @param timestamp Timestamp used in the signature
+    /// @param data Response data (a `uint256` encoded in contract ABI)
+    /// @param signature Template ID, a timestamp and the response data signed
+    /// for this contract by the Airnode address
+    function updateBeaconWithDomainSignedData(
+        address airnode,
+        bytes32 templateId,
+        uint256 timestamp,
+        bytes calldata data,
+        bytes calldata signature
+    ) external override onlyValidTimestamp(timestamp) {
+        require(
+            (
+                keccak256(
+                    abi.encodePacked(
+                        block.chainid,
+                        address(this),
+                        templateId,
+                        timestamp,
+                        data
+                    )
+                ).toEthSignedMessageHash()
+            ).recover(signature) == airnode,
+            "Signature mismatch"
+        );
+        bytes32 beaconId = deriveBeaconId(airnode, templateId);
+        uint256 decodedData = processBeaconUpdate(beaconId, timestamp, data);
+        emit UpdatedBeaconWithSignedData(beaconId, decodedData, timestamp);
```

Comment:
Would it be useful in the future to emit a different event than the one emitted in the `updateBeaconWithSignedData()` function?

Reply:
I thought about that, but all the infrastructure built to listen to these updates will now also need to be aware of that even though it doesn't matter to them. I suppose it's worth it for the sake of consistency.